### ISSUE TITLE
statement_indentation is indenting already correct line

### ISSUE THREAD

**Issue by VincentLanglet** [PHP-CS-Fixer/PHP-CS-Fixer#7884]
### Description

When running `statement_indentation` on a multiline if, if the first line is missing indentation, the following lines are getting useless extra indentation. This is especially a blocking point for https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7813 since
```php
if (true
    || false
) {
    // foo
}
```
is fixed to
```php
if (
true
    || false
) {
    // foo
}
```
and then to
```php
if (
    true
        || false // This line is indented for no reason
) {
    // foo
}
```

When I expect
```php
if (
    true
    || false // This line is untouched
) {
    // foo
}
```

### Runtime version

`3.51.0`

### Configuration file

`'statement_indentation' => true`

### Code snippet that reproduces the problem

  ```php
  <?php
if (
true
    || false
) {
    // foo
}
  ```

**Comment by VincentLanglet** [comment]
In the same way, I would expect
```
if (
    true
|| false
) {
    // foo
}
```
to be fixed to
```
if (
    true
    || false
) {
    // foo
}
```
but the code is unchanged.

**Comment by julienfalque** [comment]
Yes, I introduced it and made most of the bugfixes since then. I should take time to add comments and maybe refactor things to make it more clear. I'm having a look at this bug for now :)

**Comment by VincentLanglet** [comment]
Still relevant, waiting for https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7901
